thriftpy2/http.py

```diff
@@ -282,6 +282,7 @@ def make_client(service, host='localhost', port=9090, path='', scheme='http',
                 proto_factory=TBinaryProtocolFactory(),
                 trans_factory=TBufferedTransportFactory(),
                 ssl_context_factory=None,
+                custom_headers=None,
```

Comment:
I think a `http_headers_factory` is more suitable since we can specify headers per request.

Reply:
It took me a minute to figure out what you were referring to, but I got it now and am working through that. After thinking about it I realized that we actually need exactly what you are referring to.

I am trying to figure out how to implement test cases around this feature, but am struggling a little since it looks like the HTTP request is sort of hidden in the API, so if you have some pointers that would be great.

I will update the PR once I get an implementation put together.

Reply:
@aisk I have updated this PR to provide a header factory as suggested
@ethe I updated test_http.py to add test cases for the different ways that the factory can be used

Let me know if there is anything else

Reply:
Sorry @ermeaney I missed the comment here ... The codes you updated in this PR is just the same as I think, I think it's ok, just a few small comments here.

Reply:
@aisk I resolved the things you pointed out in the comments, and thanks for the pointers